### PR TITLE
fix(keyboard): navigate files in display order (Issue #261)

### LINKED ISSUE #261: [Keyboard Navigation] Files selected out of order when using w/s keys
## Description

When navigating files using the w (previous file) and s (next file) keyboard shortcuts, files are selected in an unexpected order that doesn't match the order they appear in the file list.

## Steps to Reproduce

1. Open a pull request with multiple file changes
2. Press 's' to navigate to the next file
3. Press 'w' to navigate to the previous file
4. Observe the selected file jumping around rather than moving sequentially through the file list

## Expected Behavior

- Pressing 's' should select the next file in the visible file list (in alphabetical order)
- Pressing 'w' should select the previous file in the visible file list (in alphabetical order)
- Navigation should follow the same order as displayed in the file list sidebar

## Actual Behavior

Files are being selected in an inconsistent order that doesn't match their display order.

## Root Cause

In `src/features/diff/hooks/useIterationAwareFiles.ts`, the files array is sorted alphabetically by filename (line 195):

```typescript
iterationFiles.sort((a, b) => a.filename.localeCompare(b.filename));
```

However, the `originalIndex` values still reference the file's position in the original GitHub API list. When `useKeyboardShortcuts.ts` navigates files, it finds the current file using `files.findIndex((f) => f.originalIndex === selectedFileIndex)` (lines 31, 47), which operates on the sorted array, but the indices don't correspond to sequential positions after sorting.

## Affected Code

- `src/features/keyboard/hooks/useKeyboardShortcuts.ts` (lines 20-60: selectNextFile and selectPreviousFile callbacks)
- `src/features/diff/hooks/useIterationAwareFiles.ts` (line 195: alphabetical sorting)

## Additional Context

This is part of the keyboard navigation feature implementation (feature/keyboard-navigation branch).

### PR DESCRIPTION
**Pull request by pedropaulovc**
## Summary
- Keyboard navigation (w/s keys) now follows the same order as the visual file list display
- Files grouped by folder match the visual order in FileList
- Creates `useFileDisplayOrder` hook that mirrors FileList grouping logic

## Root Cause
The bug was that `useKeyboardShortcuts` used `useIterationAwareFiles` (alphabetically sorted) while `FileList` groups files by folder before displaying. When pressing 's' to navigate to next file, it would jump to the wrong file because the navigation order didn't match the visual order.

## Fix
Created `useFileDisplayOrder` hook that applies the same folder grouping as FileList, then updated `useKeyboardShortcuts` to use this new hook.

Fixes #261

## Test plan
- [x] Unit tests updated to verify navigation follows display order
- [x] Manual testing confirmed navigation matches visual order
- [x] All existing tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

🔍 **[Review in CodjiFlo](https://codjiflo.vza.net/pedropaulovc/codjiflo/267)**

### CODE REVIEW

File: src/features/diff/hooks/useFileDisplayOrder.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    const grouped = groupFilesByFolder(iterationFiles);
+    return flattenGroupedFiles(grouped);
+  }, [iterationFiles]);
+
+  return {
+    files: filesInDisplayOrder,
+    isIterationMode,
+    totalFilesInPR,
+  };
+}
```
[the remark this answers is not in the document]
Comment (pedropaulovc) [reply src/features/diff/hooks/useFileDisplayOrder.ts]: Fixed in e772150 - added comprehensive unit tests for file-grouping utilities in `src/features/diff/utils/file-grouping.test.ts` covering:

1. Files grouped correctly by parent folder
2. Folders sorted alphabetically with root "/" first
3. Files within each folder maintain their order
4. Empty file arrays handled correctly
5. Edge cases like root-level files and nested folders

13 new tests added.

-- 🤖 [Claude Code](https://claude.ai/claude-code)
